cachedb/redis.c

```diff
@@ -249,16 +271,27 @@ redis_lookup(struct module_env* env, struct cachedb_env* cachedb_env,
 
 static void
 redis_store(struct module_env* env, struct cachedb_env* cachedb_env,
-	char* key, uint8_t* data, size_t data_len)
+	char* key, uint8_t* data, size_t data_len, time_t ttl)
 {
 	redisReply* rep;
-	char cmdbuf[4+(CACHEDB_HASHSIZE/8)*2+3+1]; /* "SET " + key + " %b" */
 	int n;
+	int set_ttl = (int)(env->cfg->redis_set_ttl && (!env->cfg->serve_expired || env->cfg->serve_expired_ttl > 0));
+	char cmdbuf[4+(CACHEDB_HASHSIZE/8)*2+3+((2+11)*set_ttl)+1]; /* "SETEX " + key + " " + ttl + " %b" */ 
```

Comment:
There are 2 commands that we need to support (I use underscores for spaces):
- `SET_<KEY>_%b\0`
  4 + (CACHEDB_HASHSIZE/8)*2 + 3 +1
- `SETEX_<KEY>_<TTL>_%b\0`
  6 + (CACHEDB_HASHSIZE/8)*2 + 11 + 3 +1

I believe that the correct value to use here is the second one (longer one) unless I miss something.

Reply:
`((2+11)*set_ttl)`

this basically added the required additional bytes to the string size. As it is obviously easy to miss do you think I should write an if/else block?

Reply:
To clarify my comment, your calculation is correct but it is better/clearer to use the longer one (in case we want to expand this further). Having enough space for the command is what we care most.

Reply:
.. and you can put as a comment above which commands are currently supported.

Reply:
will do